File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline

def command(x):
    if x[0] == 'push':
        arr.append(x[1])
    elif x[0] == 'pop':
        if len(arr) == 0:
            print(-1)
        else:
            print(arr.pop())
    elif x[0] == 'size':
        print(len(arr))
    elif x[0] == 'empty':
        if len(arr) == 0:
            print(1)
        else:
            print(0)
    elif x[0] == 'top':
        if len(arr) == 0:
            print(-1)
        else:
            print(arr[-1])
            
n = int(input())
arr = []
for _ in range(n):
    command(list(input().split()))

File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {

    public static void main(String[] args) throws IOException {
    	BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    	Stack<Integer> stack = new Stack<>();
    	
    	int n = Integer.parseInt(br.readLine());
    	for (int i=0; i<n; i++) {
    		String input = br.readLine();
    		if (input.contains("push")) {
    			String[] parts = input.split(" ");
    			stack.push(Integer.parseInt(parts[1]));
    		} else if (input.equals("pop")) {
    			System.out.println(stack.size()>0 ? stack.pop() : -1);
    		} else if (input.equals("size")) {
    			System.out.println(stack.size());
    		} else if (input.equals("empty")) {
    			System.out.println(stack.size()>0 ? 0 : 1);
    		} else if (input.equals("top")) {
    			System.out.println(stack.size()>0 ? stack.peek() : -1);
    		}
    	}
    }
}
